test/e2e/apimachinery/validatingadmissionpolicy.go

```diff
@@ -371,6 +376,415 @@ var _ = SIGDescribe("ValidatingAdmissionPolicy [Privileged:ClusterAdmin]", func(
+				}
+				if vapWatched.Annotations["patched"] == "true" {
+					sawAnnotation = true
+					vapWatch.Stop()
+				} else {
+					framework.Logf("missing expected annotations, waiting: %#v", vapWatched.Annotations)
+				}
+			case <-time.After(wait.ForeverTestTimeout):
+				framework.Fail("timed out waiting for watch event")
+			}
+		}
+
+		ginkgo.By("getting /status")
+		resource := admissionregistrationv1.SchemeGroupVersion.WithResource("validatingadmissionpolicies")
+		vapStatusRead, err := f.DynamicClient.Resource(resource).Get(ctx, vapCreated.Name, metav1.GetOptions{}, "status")
+		framework.ExpectNoError(err)
+		gomega.Expect(vapStatusRead.GetObjectKind().GroupVersionKind()).To(gomega.Equal(admissionregistrationv1.SchemeGroupVersion.WithKind("ValidatingAdmissionPolicy")))
+		gomega.Expect(vapStatusRead.GetUID()).To(gomega.Equal(vapCreated.UID))
+
+		ginkgo.By("patching /status")
+		patchBytes = []byte(`{"status":{"conditions":[{"type":"PatchStatusFailed","status":"False","reason":"e2e","message":"Set from an e2e test","lastTransitionTime":"2024-01-01T00:00:00Z"}]}}`)
+		vapStatusPatched, err := client.Patch(ctx, vapCreated.Name, types.MergePatchType, patchBytes, metav1.PatchOptions{}, "status")
+		framework.ExpectNoError(err)
+		hasCondition := false
+		for i := range vapStatusPatched.Status.Conditions {
+			if vapStatusPatched.Status.Conditions[i].Type == "PatchStatusFailed" {
+				hasCondition = true
+			}
+		}
+		gomega.Expect(hasCondition).To(gomega.BeTrueBecause("expect the patched status exist"))
+
+		ginkgo.By("updating /status")
+		var vapStatusUpdated *admissionregistrationv1.ValidatingAdmissionPolicy
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			vap, err := client.Get(ctx, vapCreated.Name, metav1.GetOptions{})
+			framework.ExpectNoError(err)
+
+			vapStatusToUpdate := vap.DeepCopy()
+			vapStatusToUpdate.Status.Conditions = append(vapStatusToUpdate.Status.Conditions, metav1.Condition{
+				Type:               "StatusUpdateFailed",
+				Status:             metav1.ConditionFalse,
+				Reason:             "E2E",
+				Message:            "Set from an e2e test",
+				LastTransitionTime: metav1.NewTime(time.Now()),
+			})
+			vapStatusUpdated, err = client.UpdateStatus(ctx, vapStatusToUpdate, metav1.UpdateOptions{})
+			return err
+		})
+		framework.ExpectNoError(err, "failed to update status of validatingadmissionpolicy %q", vapCreated.Name)
+		hasCondition = false
+		for i := range vapStatusUpdated.Status.Conditions {
+			if vapStatusUpdated.Status.Conditions[i].Type == "StatusUpdateFailed" {
+				hasCondition = true
+			}
+		}
+		gomega.Expect(hasCondition).To(gomega.BeTrueBecause("expect the updated status exist"))
+
+		ginkgo.By("deleting")
+		err = client.Delete(ctx, vapCreated.Name, metav1.DeleteOptions{})
+		framework.ExpectNoError(err)
+		_, err = client.Get(ctx, vapCreated.Name, metav1.GetOptions{})
+		if !apierrors.IsNotFound(err) {
+			framework.Failf("expected 404, got %#v", err)
+		}
```

Comment:
this will fail if something in the cluster adds finalizers to VAP objects. I'd suggest letting this succeed if the Get returns a 404 OR an object with non-nil deletionTimestamp + finalizers